File: day2/src/Part2Test.java
import static org.junit.jupiter.api.Assertions.*;

class Part2Test {

    @org.junit.jupiter.api.Test
    void calculateDivisibleValuesOnLine() {
        // Arrange
        Part2 p2 = new Part2();

        // Act
        int checksum1 = p2.calculateDivisibleValuesOnLine("5\t9\t2\t8");
        int checksum2 = p2.calculateDivisibleValuesOnLine("9\t4\t7\t3");
        int checksum3 = p2.calculateDivisibleValuesOnLine("3\t8\t6\t5");

        // Assert
        assertEquals(4, checksum1);
        assertEquals(3, checksum2);
        assertEquals(2, checksum3);
    }
}

File: day3/src/Part2Test.kt
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class Part2Test {

    @Test
    fun calculate() {
        // Arrange
        val testMap = HashMap<Int, Int>()
        // target, expectedResult
        testMap.put(4, 5)
        testMap.put(23, 25)
        testMap.put(27, 54)
        testMap.put(747, 806)

        val part2 = Part2()

        // Act
        testMap.forEach() {
            part2.map.clear()
            val result = part2.calculate(it.key)

            // Assert
            kotlin.test.assertEquals(it.value, result)
        }
    }
}